.scripts/generateSdks.ts

```
@@ -67,49 +46,75 @@ export async function findMissingSdks(azureRestApiSpecsRepository: string): Prom
+            const missingSdk = await analyzeSingleDirectory(fullServicePath, serviceDirectory, sdkTypeDirectory);
+            if (missingSdk) {
+                missingSdks.push(missingSdk);
             }
+        }
+    }
 
-            const readmeFiles = (await fs.readdir(fullSdkPath)).filter(file => /^readme/.test(file));
-            const fullSpecName = `${serviceDirectory} [${sdkTypeDirectory}]`
-            const sdk = { sdkName: serviceDirectory, sdkType: sdkTypeDirectory };
-
-            if (readmeFiles.length <= 0) {
-                // No readme.md
-                continue;
-            } else if (arrayContains(readmeFiles, "readme.nodejs.md")) {
-                if (!arrayContains(readmeFiles, "readme.typescript.md")) {
-                    missingSdks.push(sdk);
-                    _logger.logWithDebugDetails(`${fullSpecName}`.negative, "readme.nodejs.md exists but no matching readme.typescript.md");
-                } else {
-                    _logger.logDebug(fullSpecName.positive);
-                }
-            } else if (arrayContains(readmeFiles, "readme.md")) {
-                const readmeMdPath = path.resolve(fullSdkPath, "readme.md");
-                if (await doesReadmeMdFileSpecifiesTypescriptSdk(readmeMdPath)) {
-                    missingSdks.push(sdk);
-                    _logger.logWithDebugDetails(`${fullSpecName}`.negative, "typescript mentioned in readme.md but no readme.typescript.md exists");
-                } else {
-                    _logger.logDebug(fullSpecName.positive);
-                }
+    return missingSdks;
+}
+
+async function findDirectoriesRecursively(directoryPath: string): Promise<string[]> {
+    const foo = async (dirPath: string, paths: string[]) => {
+        const children = await getChildDirectories(dirPath);
+        for (const child of children) {
+            const fullPath = path.resolve(dirPath, child);
+            if (!(await isDirectory(fullPath))) {
+                return;
```

Why are you checking if fullPath is a directory when it is a cominbation of `dirPath` and an element of the `getChildDirectories()` function result? Shouldn't that mean that it is always a directory? If this is an error scenario, then you should be throwing instead of just silently returning.